### PR TITLE
doc(zjow): add system design doc for agent

### PR DESCRIPTION
**Pull request by zjowowen**
Add system design doc for DI-engine RL agent.

### CODE REVIEW

File: source/03_system/agent_zh.rst [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+关于RL Agent
+-------------------------------
+DI-engine的RL Agent类，顾名思义，是用于RL训练或评估过程中与环境进行直接交互的智能体。
+
+初始化后，每个RL Agent都会维护一个环境实例，一个策略实例，及其对应的相关配置。
+
+DI-engine的每个RL算法将会被实现为特定类型的RL Agent类。每个类都配备了标准的训练与评估方法，以确保同一种算法在不同环境下都有相同的标准训练与评估流程。
+
+RL Agent类支持包括Gym Atari和Mujoco在内的一系列经典RL基准环境，确保了类对于这些多样环境的兼容性。
```
Comment (PaParaZz1) [inline source/03_system/agent_zh.rst:15]: MuJoCo

File: source/03_system/agent_zh.rst [hunk 2]
```diff
@@ -0,0 +1,56 @@
+Agent
+===============================
+
+.. toctree::
+   :maxdepth: 2
+
+关于RL Agent
+-------------------------------
+DI-engine的RL Agent类，顾名思义，是用于RL训练或评估过程中与环境进行直接交互的智能体。
```
Comment (PaParaZz1) [inline source/03_system/agent_zh.rst:9]: 英文前后空格

File: source/03_system/agent_zh.rst [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+.. image::
+    images/agent.png
+    :width: 684
+    :align: center
+
+
+RL Agent类为RL的各类训练、评估管线提供了有效的抽象。
+
+它包括四个主要方法："train"、"deploy"、"batch_evaluate"和"collect_data"。这让四种不同的DI-engine管线合并为一个包含四个方法的类。这避免了之前需要在不同的文件中编写个别流水线代码的需求。
```
Comment (PaParaZz1) [inline source/03_system/agent_zh.rst:25]: ``train``

**Review by PaParaZz1 — CHANGES_REQUESTED** [review]

File: source/03_system/agent_zh.rst (hunk 1 above)
Comment (zjowowen) [reply source/03_system/agent_zh.rst]: fixed.

File: source/03_system/agent_zh.rst (hunk 2 above)
Comment (zjowowen) [reply source/03_system/agent_zh.rst]: fixed.

File: source/03_system/agent_zh.rst (hunk 3 above)
Comment (zjowowen) [reply source/03_system/agent_zh.rst]: fixed.

**Review by PaParaZz1 — APPROVED** [review]

**Pull request merged**